common/types/block_test.go

```diff
@@ -44,25 +44,19 @@ func TestStringToNodeID(t *testing.T) {
 	pubkey := genByte32()
 	nodeID1 := NodeID{
 		Key:          util.Bytes2Hex(pubkey[:]),
-		VRFPublicKey: []byte("22222"),
+		VRFPublicKey: []byte(nil),
 	}
 	nodeIDStr := nodeID1.String()
 	reversed, err := StringToNodeID(nodeIDStr)
 
 	r := require.New(t)
 	r.NoError(err, "Error converting string to NodeID")
 	r.Equal(nodeID1.Key, reversed.Key, "Node ID deserialization Key does not match")
-	r.Equal(nodeID1.VRFPublicKey, reversed.VRFPublicKey, "Node ID deserialization VRF Key does not match")
 
 	// Test too short
 	_, err = StringToNodeID(string(pubkey[:10]))
 	r.Error(err, "Expected error converting too-short string to NodeID")
 
-	// Test too long
```

Comment:
You should leave a "too long" test for two reasons:

- if we're really ignoring everything after the first 32 bytes, then you should test passing in > 32 bytes and making sure you only get the first 32 back
- we want a regression test, i.e., one that would've failed before this fix but passes now

Reply:
Makes sense, I added in the too-long tests for both the string and bytes deserialization